lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/security/SystemAccessControlFactory.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.base.security;
+
+import com.google.common.collect.ImmutableList;
+import io.trino.spi.security.SystemAccessControl;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.regex.Pattern;
+
+import static io.trino.plugin.base.security.CatalogAccessControlRule.AccessMode.ALL;
+import static java.util.Objects.requireNonNull;
+
+public class SystemAccessControlFactory
+{
+    private final FileBasedSystemAccessControlRules rules;
```

Comment:
We could have a preparatory commit which could rename this class to `SystemAccessControlRules`

Reply:
refactored